src/main.rs

```diff
@@ -1,10 +1,26 @@
-use std::io::{ Stdin, Stdout, Write };
+use std::io::{Error, Stdin, Stdout, Write};
 
 fn main() {
+    let terminal = Terminal::new();
+    let result = start_todo(terminal);
+
+    if let Err(error) = result {
+        println!(
+            "Ocorreu um erro, contacte o administrador. Erro: {:?}",
+            &error
+        )
+    }
+}
+
+fn start_todo(mut terminal: Terminal) -> Result<(), TerminalError> {
     loop {
-        let mut terminal1 = Terminal::new();
-        let todo = terminal1.ask_for_new_todo();
-        terminal1.show_todo(&todo);
+        let todo = terminal.ask_for_new_todo()?;
+        match todo {
+            Some(value) => {
+                _ = terminal.show_todo(&value);
+            }
+            None => return Ok(()),
+        }
```

Comment:
Esse `match` pode ficar um pouco mais simples sem o `_`: 
```suggestion
        match todo {
            Some(value) =>  terminal.show_todo(&value)?,
            None => return Ok(()),
        }
```

Reply:
Verdade, reduz linha e ainda melhora a visualização. 🙃